### DISCUSSION TITLE
Nothing to boot, no more network devices

### DISCUSSION THREAD

**Discussion by sqsous** [cobbler/cobbler#3716] [Q&A]
Hello,

I have installed cobbler on centos9 and I have followed these documentations 
https://cobbler.readthedocs.io/en/latest/quickstart-guide.html
https://www.alibabacloud.com/blog/how-to-install-and-configure-cobbler-on-ecs-centos-7_595708
https://www.linuxtechi.com/install-and-configure-cobbler-on-centos-7/

cobbler, dhcp and tftp are running with no issues 

cobbler has been configured to manage dhcp and dns with next server ip set to 10.0.2.9 
dhcp is configure to the following 
![Screenshot 2024-06-09 at 6 44 53 PM](https://github.com/cobbler/cobbler/assets/84263110/100202b4-bf3e-481c-9b4a-0c74039b32db)

and this is what my bare-metal shows ( my virtual machine which is configured to boot network first )
![Screenshot 2024-06-09 at 6 45 54 PM](https://github.com/cobbler/cobbler/assets/84263110/527d981e-cd8d-4dbe-87c5-070087a0ff8d)

I have looked but couldn't see or understand what is going wrong 

tftp has all the correct grub folders after running cobbler mkloaders 
and yes I did ran cobbler sync after every change 

Please help and thanks!!

**Comment by sqsous** [comment]
Also: I am trying to boot Fedora 17, and I did attach the a ks file to the profile with URL which I got from cobbler profile report

**Comment by sqsous** [comment]
Okay I managed to bypass the first issue but now getting this 
![Screenshot 2024-06-09 at 10 13 38 PM](https://github.com/cobbler/cobbler/assets/84263110/40641ffc-0620-49c3-b247-585d01c2054d)
I confirmed with tftp get /grub/grub.0 that the file exists and check it manually myself that its there

**Comment by sqsous** [comment]
Also getting this error now 
![Screenshot 2024-06-09 at 10 27 29 PM](https://github.com/cobbler/cobbler/assets/84263110/23652194-c2b4-4eeb-9d44-feec0fb02bde)
it can't connect to the host to get the kickstart file 
and I didn't use --kickstart when editing the profile, I used --autoinstall 
because whenever I tried to use kickstart it kept saying that such option doesn't exists
cobbler version 3.2.2

**Reply by SchoolGuy** [reply]
So the curl error says that this is a network-level error. If the host cannot be contacted then the kernel options for the network are incorrect or there is a firewall in between the VM/server and Cobbler.

**Reply by sqsous** [reply]
@SchoolGuy thank you! yes I figured it was firewall issues 
but there is a thing, I installed cobbler from scratch again and this issue happened 
Nothing to boot: No such file or directory 
I have checked DHCP, the config is correct, did cobbler sync 
I am at lost, any help would be appreciated

**Reply by SchoolGuy** [reply]
Nothing to boot seems to be a client issue. It normally signals that the EFI doesn't know which device it should choose for boot.

If this message is coming directly from the network card EFI then this is an error message I have not seen, normally they emit different messages that point you towards the error a bit more direct.

**Reply by sqsous** [reply]
Apparently why it wasn't working is because I also had to put true for manage_dhcp_v4.
One more question, I have configured the profile and distro as following 
![image](https://github.com/cobbler/cobbler/assets/84263110/a1e0b48a-7b78-499a-8c5b-d727d1088555)
![image](https://github.com/cobbler/cobbler/assets/84263110/367a2a0e-67da-4a65-9a83-3dcb0a3a1b1c)
![image](https://github.com/cobbler/cobbler/assets/84263110/1eaeaa47-e4cc-4357-82d1-deb3ad5adcd2)
and I have tried both the seed and ks files 
yet every time it tried to install the OS from https://ubuntu..... links 
I couldn't find any documentation to understand why its not taking the link from either the auto install file or from the distro 
Thank you ( and for your patience :D )

**Reply by SchoolGuy** [reply]
I am confused on why you would use a kickstart file for installation of Ubuntu but the interesting part here is I guess the URL that is given to the kernel command line in the grub config for the profile or system.

**Reply by SchoolGuy** [reply]
Also what is the output of `cobbler profile get-autoinstall <profile name>`?

**Reply by sqsous** [reply]
Because when it didn't work, I thought adding kickstart was the missing part
![image](https://github.com/cobbler/cobbler/assets/84263110/e1ba8041-6faf-441d-9bf3-7b989750c400)

**Reply by SchoolGuy** [reply]
That looks wrong to me. Can you please switch to a proper valid preseed template? Also please provide the output of the GRUB config file in the TFTP directory (should be `<arch>_menu_items.cfg` if I am not mistaken).

**Reply by sqsous** [reply]
![image](https://github.com/cobbler/cobbler/assets/84263110/edc340ef-b59d-40db-a135-d9d341d8a158)![image](https://github.com/cobbler/cobbler/assets/84263110/9f46e0c5-acc4-4b6a-9403-132eff20f965)

these one? its for x86_64_menu_items.cfg

**Reply by sqsous** [reply]
I tried to make my preseed template as simple as possible, to be able to debug why the URL is not working

**Reply by SchoolGuy** [reply]
That looks sound to me. Are Cobbler and the target machine on the same OSI layer 2 domain or could it be that a firewall prevents access to the URL?

From the network where the machine will be installed can you do a `curl http://10.0.2.28/cblr/svc/op/autoinstall/profile/ubuntu-casper-x86_64`? If that works then Cobbler works as expected I would say. Since you have `netcfg/choose_interface=auto` it could be that the installer uses the wrong network interface.

**Reply by sqsous** [reply]
I tried from different vm in the same network 
<img width="863" alt="image" src="https://github.com/cobbler/cobbler/assets/84263110/53d1e39f-2903-4688-b1c0-5e66a8b9374b">
could it also be an issue than this x86_64_menu_items.cfg is not defined as a filename in DHCP template in cobbler ?

**Reply by SchoolGuy** [reply]
Okay that the `curl` works is very good news. That means that the installer could fetch the file.

Are you erroring out in iPXE, GRUB or the installer?

**Reply by sqsous** [reply]
<img width="639" alt="image" src="https://github.com/cobbler/cobbler/assets/84263110/4c9e90bc-9110-4f2e-83ef-32058284e838">
![image](https://github.com/cobbler/cobbler/assets/84263110/ee8ad3e3-219d-4df6-a764-f15c538f07ae)
<img width="558" alt="image" src="https://github.com/cobbler/cobbler/assets/84263110/c841a07d-fca8-4500-81c1-76a058425aa4">
<img width="640" alt="image" src="https://github.com/cobbler/cobbler/assets/84263110/a3f22cd7-25de-46d7-93a6-24f9b56cfe11">
<img width="642" alt="image" src="https://github.com/cobbler/cobbler/assets/84263110/6ce08adc-9576-490e-bab6-fe413f92546a">
so I guess the issue is in the installer or grub, because it doesn't want to use the URL defined in the x86_64_menu_items.cfg

**Reply by sqsous** [reply]
<img width="643" alt="image" src="https://github.com/cobbler/cobbler/assets/84263110/cc25cdf9-91e7-4baf-a14c-45429a6973cf">

**Reply by SchoolGuy** [reply]
So grub works fine and the kernel and initrd are correctly downloaded. That is good.

The issue is that the the installer tries to do an install with the CD-ROM.

Debian has special Netinstall media, I believe you have used the standard install medium.

**Reply by sqsous** [reply]
is there a way to check that ? 
![image](https://github.com/cobbler/cobbler/assets/84263110/17984bd2-aa65-4876-af29-0d8c938ca539)
because in here it just shows that it is taking from distro_mirror

**Reply by SchoolGuy** [reply]
Here is a manual from Ubuntu: https://ubuntu.com/server/docs/how-to-netboot-the-server-installer-on-amd64

Under "Mode independent setup" you find the correct image that you need. (Link will need to be adjusted for your OS version.)

**Reply by sqsous** [reply]
<img width="641" alt="image" src="https://github.com/cobbler/cobbler/assets/84263110/51682eb3-5ca8-4015-8805-d49f41568d00">
hmm interesting, it is still giving the same error 
( I used [ubuntu-22.04.4-live-server-amd64.iso](https://releases.ubuntu.com/jammy/ubuntu-22.04.4-live-server-amd64.iso) )
as if its still not seeing the preseed not the config in x86_64_menue_items.cfg
I manually, during the installation, tried to pass one time the URL defined in cobbler distro report and second time using the URL defined in x86_64_menue_items.cfg

**Reply by SchoolGuy** [reply]
You do see the pressed file being downloaded in the picture you sent. I am not sure if this was the attempt where you manually edited the config file.

I can try next weekend to reproduce the installation of Ubuntu, normally I only need to boot openSUSE/SLES.

**Reply by sqsous** [reply]
It was downloaded because after the kernel and initrd were downloaded correctly, the installation required me to enter yes  if I want to install using URL and then other options ( shown In the screenshot )
![image](https://github.com/cobbler/cobbler/assets/84263110/2902e842-8053-4ea0-b463-3fda64f161cd)

Thank you! that  would be amazing if you can try :)

**Reply by SchoolGuy** [reply]
Okay so Ubuntu 22.04 doesn't provide the Debian installer files and as such won't work with Cobbler. I wasn't aware of this change in the Ubuntu community. The last version having the legacy installer is Ubuntu 18.04.

Link to the legacy installer images: https://cdimage.ubuntu.com/netboot/

Sadly we don't support cloud-init just yet. To follow updates please keep in touch with issue #3218. If you absolutely must use a recent version of Ubuntu please follow the community guide in #3573 to have a temporary workaround.

I got Ubuntu 18.04 and Fedora 17 working with the steps I posted in a blog on the project website just for you. It was half a day of work, so I really hope that this helps. I wasn't able to encounter any bugs that prevented me from progressing, just the normal adjustments to the network and host OS Cobbler is running on.

Article: https://cobbler.github.io/blog/2024/06/30/libvirt-ipxe.html

**Reply by sqsous** [reply]
Thank you for the blog and for the links! I will try out the temp workaround! 
Thanks again for the time and help :)
